src/components/Messages/Message.tsx

```diff
@@ -51,7 +51,7 @@ const Message: FC<MessageProps> = ({ conversationKey }) => {
     }
   }, [conversationKey, hasMore, messages, endTime]);
 
-  if (!currentProfile || !isFeatureEnabled('messages', currentProfile.id)) {
+  if (!isFeatureEnabled('messages', currentProfile?.id)) {
```

Comment:
I personally preferred the previous check because I think we want to 404 there even after the feature flag is removed if current profile doesn't exist. So I think the first version just makes it easier to strip out the feature flag code soon

Reply:
This one after disconnect will first show 404 and then route to home, if that is okay, I can add it back

Reply:
Yup its expected one